### DISCUSSION TITLE
Best way to switch vpn provider

### DISCUSSION THREAD

**Discussion by feh123** [RaspAP/raspap-webgui#1618] [General]
Hi I have a RPI4 in a wlan1 (client), wlan0 (AP) mode. I also have subscriptions for wireguard (Mullvad) and Nordvpn (dedicated IP configuration). It is useful to be able to switch between these providers as the dedicated IP is very good but sometimes causes the wifi signal to stall (I don't know why). RaspAP in this config works very well on both providers and I use it all the time. But it would be good to be able to switch.
I have never switched before on my RaspAP device and I tried this: I was running the Nordvpn, I opened the wireguard, setup the configuration file and started wireguard. The icon showed wg was on. I tried to go to the Openvpn page to turn off the Nordvpn config. But the UI just seem to freeze, I tried switching to other pages with no effect and I lost my AP. Maybe I did not wait long enough for the Nordvpn page to open? Anyway I had to reinstall RaspAP completely.
I thought I would check if there was a correct way (or better way) to switch? Thanks!

**Comment by billz** [comment]
Wireguard and OpenVPN services are not intended to be run simultaneously. Stop one service before starting the other.

**Reply by feh123** [reply]
I was just thinking I would keep a vpn active all the time but good to know the correct way. Thanks.
